Code/GraphMol/SynthonSpaceSearch/SynthonSpace.h

```diff
@@ -202,35 +206,44 @@ class RDKIT_SYNTHONSPACESEARCH_EXPORT SynthonSpace {
    * Throws a std::runtime_error if it doesn't think the format is correct,
    * which it does by checking that the first line is as above and subsequent
    * lines have appropriate number of fields.
+   * If it receives a SIGINT, returns cancelled=true.
    */
-  void readTextFile(const std::string &inFilename);
+  void readTextFile(const std::string &inFilename, bool &cancelled);
 
-  // Writes to/reads from a binary DB File in our format.
   /*!
+   * Writes to a binary DB File in our format.
    *
    * @param outFilename: the name of the file to write.
    */
   void writeDBFile(const std::string &outFilename) const;
+
   /*!
+   * Reads from a binary DB File in our format.
    *
    * @param inFilename: the name of the file to read.
    */
   void readDBFile(const std::string &inFilename);
 
-  // Write a summary of the SynthonSpace to given stream.
   /*!
+   * Write a summary of the SynthonSpace to given stream.
    *
    * @param os: stream
    */
-  void summarise(std::ostream &os) const;
+  void summarise(std::ostream &os);
 
-  // Writes the enumerated library to file in SMILES format (1 compound
-  // per line, SMILES name
   /*!
-  @param outFilename: name of the file to write
+   * Writes the enumerated library to file in SMILES format
+   * (1 compound per line, SMILES name)
+   *
+   * @param outFilename: name of the file to write
    */
   void writeEnumeratedFile(const std::string &outFilename) const;
 
+  // Below here are functions that have to be in the
```

Comment:
Q: Would it make sense to make them protected and add the other classes that need them as friends?

Reply:
It looks differently inelegant, and probably a sign of bad design but yes, a bit tidier.